### PR TITLE
Improve the help menu by grouping the flat list of commands and add `tuist project view` command

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
The help menu is hard to parse visually. Thanks to a recent improvement in Swift Argument Parser, the package that we use for parsing the CLI arguments, we can now organize the output in groups. I'm following the same grouping that we are using in the documentation for familiarity and consistency.

### How to test the changes locally 🧐
If you run the CLI locally, you should see the sections there.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -460,7 +460,7 @@ let package = Package(
         .package(url: "https://github.com/SwiftGen/StencilSwiftKit", exact: "2.10.1"),
         .package(url: "https://github.com/SwiftGen/SwiftGen", exact: "6.6.2"),
         .package(url: "https://github.com/tuist/XcodeProj", exact: "8.19.0"),
-        .package(url: "https://github.com/cpisciotta/xcbeautify", from: "2.4.0"),
+        .package(url: "https://github.com/cpisciotta/xcbeautify", branch: "main"),
```
Comment (pepicrft) [inline Package.swift:463]: @cpisciotta do you have a release planned including this change?

Comment (cpisciotta) [reply Package.swift:463]: I don't understand your question. `xcbeautify` should always be used via a semantic version tag. `main` is always "release-ready," but often doesn't include changes that should be bundled with a release. I recommend reverting this change.

Why are switching from tag to branch?

---

Separately, `2.4.1` is the latest version.

Comment (pepicrft) [reply Package.swift:463]: [This change](https://github.com/cpisciotta/xcbeautify/commit/689eb1b97806af969086e80e146a42d98956920b) is unreleased. It might have gone unnoticed because it was automated. So I was wondering if you'll prepare a 2.5.0 release including it.

File: Sources/TuistKit/Commands/TuistCommand.swift [hunk 2]
```diff
@@ -14,28 +14,35 @@ public struct TuistCommand: AsyncParsableCommand {
...
+                    GraphCommand.self,
+                    MigrationCommand.self,
+                    PluginCommand.self,
+                    RunCommand.self,
+                    ScaffoldCommand.self,
+                    TestCommand.self,
+                    CleanCommand.self,
+                    CacheCommand.self,
+                ]),
+                CommandGroup(name: "Server", subcommands: [
```
Comment (fortmarek) [inline Sources/TuistKit/Commands/TuistCommand.swift:38]: I wonder if it would be better to name this section as something like `Manage your Tuist account` instead of referring to the `Server`.

File: Sources/TuistKit/Commands/TuistCommand.swift [hunk 3]
```diff
@@ -14,28 +14,35 @@ public struct TuistCommand: AsyncParsableCommand {
...
+                    ProjectCommand.self,
+                    OrganizationCommand.self,
+                    AuthCommand.self,
+                    SessionCommand.self,
+                    AnalyticsCommand.self,
```
Comment (fortmarek) [inline Sources/TuistKit/Commands/TuistCommand.swift:43]: A bit unrelated but I wonder if we should remove this one since it only opens the server.

We could also replace it with a command similar to `gh` CLI:
`gh repo view tuist/tuist` -> `tuist project view tuist/tuist`

What do you think?

File: Sources/TuistKit/Commands/TuistCommand.swift (hunk 2 above)
Comment (danieleformichelli) [reply Sources/TuistKit/Commands/TuistCommand.swift:38]: Maybe `Account`? 🤷 :

**Review by danieleformichelli — APPROVED** [review]

File: Package.swift (hunk 1 above)
Comment (pepicrft) [reply Package.swift:463]: Thanks Charles :). I updated to 2.5.0 and the dependency resolution ran smoothly.

File: Sources/TuistKit/Commands/TuistCommand.swift (hunk 2 above)
Comment (pepicrft) [reply Sources/TuistKit/Commands/TuistCommand.swift:38]: I like `Account`. I updated it.

Comment (danieleformichelli) [reply Sources/TuistKit/Commands/TuistCommand.swift:38]: I still see `Server` :)

File: Sources/TuistKit/Commands/TuistCommand.swift (hunk 3 above)
Comment (pepicrft) [reply Sources/TuistKit/Commands/TuistCommand.swift]: Good idea! I pushed a `tuist project view` command to this PR :)
